src/api/index.js

```diff
@@ -523,3 +523,76 @@ export function getEventListener({ name, namespace }) {
   });
   return get(uri);
 }
+
+export function createTaskRun({
+  namespace,
+  taskName,
+  labels,
+  params,
+  resources,
+  serviceAccount,
+  timeout
+}) {
+  const payload = {
+    apiVersion: 'tekton.dev/v1beta1',
+    kind: 'TaskRun',
+    metadata: {
+      name: `${taskName}-run-${Date.now()}`,
+      namespace,
+      labels: {
+        'tekton.dev/task': taskName,
+        ...labels
+      }
+    },
+    spec: {
+      params: [],
+      resources: {
+        inputs: [],
+        outputs: []
+      },
+      taskRef: {
+        name: taskName
+      }
+    }
+  };
+  if (params) {
+    payload.spec.params = Object.keys(params || {}).map(name => ({
```

Comment:
no need for the `|| {}` since you're already checking it's truthy above

Reply:
Good eye. I think I added it when I copied from createPipelineRun and had a test that didn't include `params` (so bad things happened).

Since it isn't tested like that in the Pipeline function, I think it means that we assume the function always gets called with `params` and `resources` always defined, even just as empty objects? That seems reasonable to me, I just want to verify it's not an issue.